source/developer/interactive-message-buttons.rst

```diff
@@ -0,0 +1,170 @@
+
+Use these buttons to simplify complex workflows by allowing users to take quick actions directly through your integration post. For example, the buttons enable your integration to:
+
+- update sales opportunities in your CRM system from Mattermost
+- conduct a customer survey or a poll
+- file a report on market trends
+
+To try the message buttons out, you can use this `demo integration <https://github.com/mattermost/mattermost-interactive-post-demo>`_ to add polling to Mattermost channels via a `/poll` slash command.
+
+Interactive message buttons are not yet supported on Mattermost mobile apps.
+
+// XXX Add GIF image (AB preparing one for marketing)
+
+.. toctree::
+   :maxdepth: 2
+
+Button Options
+---------------
+
+Add message buttons as ``actions`` in your integration `message attachments <https://docs.mattermost.com/developer/message-attachments.html>`_.
+
+The following payload gives an example webhook that uses message buttons.
+
+.. code-block:: text
+
+  {
+    "attachments": [
+      {
+        "pretext": "This is the attachment pretext.",
+        "text": "This is the attachment text.",
+        "actions": [
+          {
+            "name": "Ephemeral Message",
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "ephemeral"
+              }
+            }
+          }, {
+            "name": "Update",
+            "integration": {
+              "url": "http://127.0.0.1:7357",
+              "context": {
+                "action": "update"
+              }
+            }
+          }
+        ]
+      }
+    ]
+  }
+
+The integration can respond with an update to the original post, or with an ephemeral message:
+
+.. code-block:: text
+
+  {
+    "update": {
+      "message": "Updated!"
+    },
+    "ephemeral_text": "You updated the post!"
+  }
+
+.. image:: ../../source/images/interactive_message.gif
+
+Below we give a brief description of each parameter to help you customize the webhook post in Mattermost. For more information on message attachments, `see our documentation <https://docs.mattermost.com/developer/message-attachments.html>`_.
+
+Name
+  Give your action a descriptive name.
+
+URL
+  The actions are backed by an integration that handles HTTP POST requests when users clicks the message button. The URL parameter determines where this action is sent to. The request contains an ``application/json`` JSON string.
+
+Context
+  The requests sent to the specified URL contain the user id and any context that was provided in the action definition. A simple example is given below:
+  
+  .. code-block:: text
+
+    {
+    "user_id": "rd49ehbqyjytddasoownkuqrxe",
+    "context": {
+      "action": "ephemeral"
```

Comment:
I would suggest changing the example to something else like `"action": "do_something"` just to avoid confusion and make it obvious that the value is user-defined and has no special meaning.

Reply:
Good point, will do